modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/SeqexecEngine.scala

```diff
@@ -92,35 +90,51 @@ class SeqexecEngine(settings: SeqexecEngine.Settings) {
     type SequenceAR = engine.Sequence[engine.Action \/ engine.Result]
     type StepAR = engine.Step[engine.Action \/ engine.Result]
 
+    def viewSequence(seq: SequenceAR, st: SequenceState): SequenceView = {
+
+      def engineSteps(seq: SequenceAR): List[Step] = {
+
+        def viewStep(step: StepAR): Step =
+          StandardStep(
+            step.config,
+            engine.Step.status(step),
+            // TODO: Implement breakpoints at Engine level
+            breakpoint = false,
+            // TODO: Implement skipping at Engine level
+            skip = false,
+            configStatus = Map.empty,
+            // TODO: Implement standard step at Engine level
+            observeStatus = ActionStatus.Pending
+          )
 
-    def viewSequence(seq: SequenceAR, st: SequenceState): SequenceView =
+        // Couldn't find this on Scalaz
+        def splitWhere[A](l: List[A])(p: (A => Boolean)): (List[A], List[A]) =
```

Comment:
I get what you mean. Actually I wrote it as a separate function because I'm pretty sure there should be something very close in Scalaz that I couldn't find it. This way it's easier to replace it later when I stumple upon it in the future.

That's usually what I do when I suspect there is a similar function already in Scalaz.